gnovm/pkg/gnomod/gnomod.go

```diff
@@ -163,6 +165,65 @@ func GnoToGoMod(f File) (*File, error) {
 	return &f, nil
 }
 
+func CreateGnoModFile(rootDir, modPath string) error {
+	if !filepath.IsAbs(rootDir) {
+		return fmt.Errorf("dir %q is not absolute", rootDir)
+	}
+
+	modFilePath := filepath.Join(rootDir, "gno.mod")
+	if _, err := os.Stat(modFilePath); err == nil {
+		return errors.New("gno.mod file already exists")
+	}
+
+	if modPath == "" {
+		// Check .gno files for package name
+		// and use it as modPath
+		files, err := ioutil.ReadDir(rootDir)
+		if err != nil {
+			fmt.Errorf("read dir %q: %w", rootDir, err)
+		}
+
+		var pkgName gnolang.Name
+		for _, file := range files {
+			if file.IsDir() || !strings.HasSuffix(file.Name(), ".gno") || strings.HasSuffix(file.Name(), "_filetest.gno") {
+				continue
+			}
+
+			fpath := filepath.Join(rootDir, file.Name())
+			bz, err := os.ReadFile(fpath)
+			if err != nil {
+				return fmt.Errorf("read file %q: %w", fpath, err)
+			}
+
+			pn := gnolang.PackageNameFromFileBody(file.Name(), string(bz))
+			if strings.HasSuffix(string(pkgName), "_test") {
+				pkgName = pkgName[:len(pkgName)-len("_test")]
+			}
+			if pkgName == "" {
+				pkgName = pn
+			}
+			if pkgName != pn {
+				return fmt.Errorf("package name mismatch: [%q] and [%q]", pkgName, pn)
+			}
+		}
+		if pkgName == "" {
+			return errors.New("cannot determine package name")
+		}
+		modPath = string(pkgName)
+	}
+
+	modFile := &File{
```

Comment:
Why not create a helper for this initialization?

Reply:
Yeah. I agree there is a helper needed for `File` initialization.
I plan to refactor code and define methods for `File`. like for this specific case `modFile.AddModuleStmt()`.
Will do it in the another PR.